### PR TITLE
Diff shows only changed lines

### LINKED ISSUE #3230: Diff should show only affected lines
Given these two strings:
```
$leExpectedXml = <<<XML
<note>
    <to>Tove</to>
    <from>Jani</from>
    <heading>Reminder</heading>
    <body>Don't forget me this weekend!</body>
</note>
XML;

$leActualXml = <<<XML
<note>
    <to>Tove</to>
    <from>Jani</from>
    <heading>Reminder
    </heading>
    <body>Don't forget me this weekend!</body>
</note>
XML;
```

and running:

```
$this->assertEquals(
    $leExpectedXml,
     $leActualXml,
     'le xmls should be the same'
);}
```

I get the output:

```
Fail  Failed asserting that two strings are equal. ( -Expected | +Actual ) 
- '<note>
-     <to>Tove</to>
-     <from>Jani</from>
-     <heading>Reminder</heading>
-     <body>Don't forget me this weekend!</body>
- </note>'
+ '<note>
+     <to>Tove</to>
+     <from>Jani</from>
+     <heading>Reminder
+     </heading>
+     <body>Don't forget me this weekend!</body>
+ </note>'
```

I expect it to look like:

```
  --- Expected 
  +++ Actual
  @@ @@
   <note>
       <to>Tove</to>
       <from>Jani</from>
  -    <heading>Reminder</heading>
  +    <heading>Reminder
  +    </heading>
       <body>Don't forget me this weekend!</body>
   </note>
```

I am going to work on a pull request for this as requested by @Naktibalda on [a related stackoverflow question](http://stackoverflow.com/questions/37858060/how-to-get-more-readable-diff-for-codeception-asserteequals-string).

### PR DESCRIPTION
**Pull request by k0pernikus**
Fixes #3230

The diff used to show both complete strings instead of only the changed lines,
I've added a unit test for the case and refactored a bit inside the Console.php

Todo: Add color output as this got lost during the diff changes.

### CODE REVIEW

**Comment by k0pernikus** [comment]
@Naktibalda The failing travis build confuses me. It seems it hangs during:

```
git clone -q -b 2.2-doctrine2 --recursive https://github.com/Naktibalda/codeception-zf2-tests frameworks-zf2
```

I don't think it's something I introduced.

**Comment by Naktibalda** [comment]
Sorry, it was my fault.
I deleted that branch because it was merged to 2.2 a while ago.
But I forgot that it is still used by Codeception .travis.yml

**Comment by raistlin** [comment]
Could you fix remaining code formatting issue Nitpick-CI is yelling about?

Seems ok.

**Comment by Naktibalda** [comment]
I will fix that Nitpick issue.

File: src/Codeception/Lib/Console/Colorizer.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * @param string $string
+     */
+    public function colorize($string = '')
+    {
+        $fp = fopen("php://memory", 'r+');
+        fwrite($fp, $string);
+        rewind($fp);
+
+        $colorizedMessage = '';
+        while ($line = fgets($fp)) {
```
Comment (Naktibalda) [inline src/Codeception/Lib/Console/Colorizer.php]: Err, what newline?

**Comment by k0pernikus** [comment]
@raistlin I alreday fixed the style in [commit 3a562a31818](https://github.com/Codeception/Codeception/pull/3242/commits/3a562a3181894bf5671f79df8b67c0f5713bfbc4). I found it odd that it wasn't marked as resolved afterwards.

@Naktibalda Thanks for merging \o/

**Comment by DavertMik** [comment]
Thanks for your work. This is just awesome!

**Pull request merged**